server/src/main/java/org/elasticsearch/snapshots/SnapshotInfo.java

```diff
@@ -700,8 +720,14 @@ public XContentBuilder toXContent(final XContentBuilder builder, final Params pa
         final boolean verbose = params.paramAsBoolean("verbose", GetSnapshotsRequest.DEFAULT_VERBOSE_MODE);
         // write snapshot info for the API and any other situations
         builder.startObject();
+        final SnapshotId snapshotId = snapshot.getSnapshotId();
         builder.field(SNAPSHOT, snapshotId.getName());
         builder.field(UUID, snapshotId.getUUID());
+
+        if (params.paramAsBoolean("include_repository", true)) {
```

Comment:
I suppose we could also leave this and related code out and just always return the repository name in responses. That would simplify code a little and adding the field seems fine even in `7.x`. We've added other fields like `metadata` in the past without user complaints or other issues. I put the flag in here for now as this is what we discussed yesterday though.

Reply:
Also +1 on adding the field by default (and adapting existing YAML tests).

Reply:
One exception though is when it is the unknown repo name, we do not want to output that.

Reply:
++ added that check